lightningd/peer_control.c

```diff
@@ -183,6 +181,87 @@ static struct io_plan *send_error(struct io_conn *conn,
 	return peer_write_message(conn, pcs, pcs->peer->error, (void *)io_close_cb);
 }
 
+struct getting_gossip_fd {
```

Comment:
I agree, this is rather a roundabout way of doing things, but the alternative of handing the peer through `gossipd` seems worse. Maybe we could desynchronize further and allow for the `gossip_fd` to be passed to `channeld` after initialization? That'd duplicate some paths too, but only in `channeld`, which would need to delay some messages.